pkg/devspace/dependency/util/util.go

```diff
@@ -142,8 +140,13 @@ func DownloadDependency(ctx context.Context, workingDirectory string, source *la
 					return "", errors.Wrap(err, "clone repository")
 				}
 			}
+
+			log.Debugf("Cloned %s", gitPath)
+		} else if !source.DisablePull {
+			repo.Pull(ctx, gitCloneOptions)
```

Comment:
If pulling is removed from `repo.Clone`, then we could change this from an `else if` to an `if`:
```golang
if !source.DisablePull && source.Revision == "" {
  // Pull
}
```

May need to also handle the error, and attempt to switching the URL, similar to [this line](https://github.com/devspace-sh/devspace/pull/2766/files#diff-a5251b488fef0b4fa9e68c8210ca97fc0553bd1507255378cb8cee3ef079fd53R129).

Reply:
I agree, take a look again.

I am not sure if me messing up the commit history matters? I hope we can squash into the origin repository.

I retested
- It correctly git clone repos when i delete .dependencies folder
- I correctly reclone git repo when a branch is changed
- It correctly git pull cached dependencies
- It respects the dependency.disablePull flag by not pulling when its true